File: lib/actions/service/add.js
'use strict';

var inquirer = require('inquirer');
var compose = require('../../compose');

module.exports = function add(name) {
  inquirer.prompt([
    {
      type: 'input',
      name: 'image',
      message: 'What will be the image source for this service?',
      default: name + ':latest',
      validate: function(image) {
        return !!image || 'An image source is required';
      }
    }
  ], function (answers) {
    compose.addService(name, {
      image: answers.image
    });

    console.log('Service ' + name + ' added');
  });
};

File: lib/frigate/commands/add.rb
require 'inquirer'
require 'colorize'
require 'frigate/compose'

module Frigate::Commands
  class Add
    def initialize(args, options)
      @compose = Frigate::Compose.new

      name = args[0]

      image = ask_image(name)

      @compose.add_service name, 'image' => image

      puts "Service #{name} added"
    end

    def ask_image(name)
      Ask.input 'What will be the image source for this service?',
                default: "#{name}:latest"
    end
  end
end
